sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/NumberConverter.scala

```diff
@@ -23,6 +23,13 @@ import org.apache.spark.unsafe.types.UTF8String
 
 object NumberConverter {
 
+  /**
+   * The output string has a max length of one char per bit in the intermediate representation plus
+   * one char for the '-' sign.  This corresponds to the representation of `Long.MinValue` with
+   * `toBase` equal to -2.
+   */
+  private final val MAX_OUTPUT_LENGTH = 64 + 1
```

Comment:
Shall we add some comment like
`Sometimes the length of intermediate representation exceeds 64, we obtain the first byte to store one char for the '-' sign.` ?

Reply:
Yeah. It's exactly 64. I made a slip of the tongue.

Reply:
Thank you for the updated comment.